src/SonarQube.Client/Services/SonarQubeClient.cs

```diff
@@ -63,45 +65,66 @@ public SonarQubeClient(HttpMessageHandler messageHandler, TimeSpan requestTimeou
                         request.OrganizationKey, request.Page, request.PageSize); // TODO: should handle optional params
 
             return InvokeSonarQubeApi(connection, query, token,
-                stringResponse => JObject.Parse(stringResponse)["components"].ToObject<ComponentDTO[]>());
+                stringResponse => JObject.Parse(stringResponse)["components"].ToObject<ComponentResponse[]>());
         }
 
-        public Task<Result<OrganizationDTO[]>> GetOrganizationsAsync(ConnectionDTO connection,
+        public Task<Result<ServerIssue[]>> GetIssuesAsync(ConnectionRequest connection, string key, CancellationToken token)
+        {
+            const string BatchIssuesAPI = "batch/issues"; // Since 5.1; internal
+
+            var query = AppendQueryString(BatchIssuesAPI, "?key={0}", key);
+
+            return InvokeSonarQubeApi(connection, query, token,
+                async (HttpResponseMessage response) =>
+                {
+                    var byteArray = await response.Content.ReadAsByteArrayAsync();
+                    // Protobuf for C# throws when trying to read outside of the buffer and ReadAsStreamAsync returns a non
+                    // seekable stream so we can't determine when to stop. The hack is to use an intermediate MemoryStream
+                    // so we can control when to stop reading.
+                    // Note we might want to use FileStream instead to avoid intensive memory usage.
+                    using (var stream = new MemoryStream(byteArray))
+                    {
+                        return ReadFromProtobufStream(stream, ServerIssue.Parser).ToArray();
+                    }
+                });
+        }
+
+        public Task<Result<OrganizationResponse[]>> GetOrganizationsAsync(ConnectionRequest connection,
             OrganizationRequest request, CancellationToken token)
         {
             const string OrganizationsAPI = "api/organizations/search"; // Since 6.2; internal
 
             var query = AppendQueryString(OrganizationsAPI, "?p={0}&ps={1}", request.Page, request.PageSize); // TODO: should handle optional params
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)